lib/rust/ensogl/component/grid-view/src/header.rs

```diff
@@ -365,6 +366,44 @@ impl<E: Entry, HeaderEntry: Entry<Params = E::Params>> GridView<E, HeaderEntry>
     }
 }
 
+impl<E: Entry, HeaderEntry: Entry<Params = E::Params>> FrpNetworkProvider
+    for GridView<E, HeaderEntry>
+{
+    fn network(&self) -> &frp::Network {
+        self.model.grid.network()
+    }
+}
+
+impl<E: Entry, HeaderEntry: Entry<Params = E::Params>> application::View
+    for GridView<E, HeaderEntry>
+{
+    fn label() -> &'static str {
+        "GridViewWithHeaders"
+    }
+
+    fn new(app: &Application) -> Self {
+        GridView::<E, HeaderEntry>::new(app)
+    }
+
+    fn app(&self) -> &Application {
+        self.model.grid.app()
+    }
+
+    fn default_shortcuts() -> Vec<application::shortcut::Shortcut> {
+        use application::shortcut::ActionType::*;
+        (&[
+            (PressAndRepeat, "up", "move_selection_up"),
+            (PressAndRepeat, "down", "move_selection_down"),
+            (PressAndRepeat, "left", "move_selection_left"),
+            (PressAndRepeat, "right", "move_selection_right"),
+        ])
+            .iter()
+            .map(|(a, b, c)| Self::self_shortcut_when(*a, *b, *c, "focused"))
```

Comment:
You placed this definition just below definition of the struct. This means that this is the most important code this struct has. Is this intentional? I believe that struct methods are more imoportant. Look at other codebases, e.g. the graph editor, to see what is the coe layout and where we are definining shortcuts.